src/st/dai/e_bone_pillar.c

```diff
@@ -0,0 +1,489 @@
+                offsetY += ((rsin(angle) * 0xE) >> 0xC);
+                g_api.CheckCollision(offsetX, offsetY, &collider, 0);
+                if (collider.effects & EFFECT_SOLID) {
+                    PlaySfxPositional(SFX_SKULL_KNOCK_B);
+                    self->posY.i.hi += collider.unk18;
+                    self->velocityY = -self->velocityY;
+                    self->velocityY -= self->velocityY / 2;
+                    break;
+                }
+            }
+            if (self->flags & FLAG_DEAD) {
+                PlaySfxPositional(SFX_QUICK_STUTTER_EXPLODE_B);
+                self->hitboxState = 0;
+                self->drawFlags = FLAG_DRAW_DEFAULT;
+                self->step = 0;
+                self->pfnUpdate = EntityExplosion;
+                self->params = 1;
+            }
+        }
+        break;
+    case 0xFF:
+#include "../pad2_anim_debug.h"
+    }
+}
+
+void EntityBonePillarSpikeBall(Entity* self) {
+    Collider collider;
+    Entity* entity;
+    Primitive* prim;
+    s32 primIndex;
+    s32 posX, posY;
+
+    if (self->flags & FLAG_DEAD) {
+        PlaySfxPositional(SFX_QUICK_STUTTER_EXPLODE_B);
+        entity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+        if (entity != NULL) {
+            CreateEntityFromEntity(E_EXPLOSION, self, entity);
+            entity->params = 3;
+        }
+        DestroyEntity(self);
+        return;
+    }
+    switch (self->step) {
+    case 0:
+        InitializeEntity(g_EInitBonePillarSpikeBall);
+        self->animCurFrame = 0x16;
+        self->hitboxWidth = 0xD;
+        self->hitboxHeight = 0xD;
+        self->flags |= FLAG_UNK_02000000;
+        primIndex = g_api.AllocPrimitives(PRIM_GT4, 1);
+        if (primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+        self->flags |= FLAG_HAS_PRIMS;
+        self->primIndex = primIndex;
+        prim = &g_PrimBuf[primIndex];
+        prim->tpage = self->unk5A / 4;
+        if (self->unk5A & 1) {
+            posX = 0x80;
+        } else {
+            posX = 0;
+        }
+        if (self->unk5A & 2) {
+            posY = 0x80;
+        } else {
+            posY = 0;
+        }
+        prim->clut = g_EInitBonePillarSpikeBall[3] + 2;
+        prim->u0 = prim->u2 = posX + 0x50;
+        prim->u1 = prim->u3 = posX + 0x70;
+        prim->v0 = prim->v1 = posY + 0x30;
+        prim->v2 = prim->v3 = posY + 0x50;
+        prim->priority = self->zPriority - 1;
+        prim->drawMode = DRAW_UNK02;
+    case 1:
+        if (self->ext.et_bonePillar.unk85) {
+            self->drawFlags = FLAG_DRAW_ROTATE;
+            self->hitPoints = 0x7FFF;
+            self->hitboxState = 1;
+            self->velocityX = FIX(-0.75);
+            self->velocityY = FIX(-1.0);
+            self->step++;
+            entity = AllocEntity(&g_Entities[32], &g_Entities[47]);
+            if (entity != NULL) {
+                DestroyEntity(entity);
+                entity->entityId = E_SEALED_DOOR;
```

Comment:
I feel like this should instead be a second spiked ball E_ID, rather than E_SEALED_DOOR.  Seems like something may be slightly off with the E_IDs in dai.h